R/plotTemporal.R

```diff
@@ -0,0 +1,145 @@
+  if(identical(class(storm), "character")){
+    stopifnot("storm not find in data" = (storm %in% names(data)))
+  }else if(identical(class(storm), "numeric")){
+    stopifnot("storm index out of bound in data" = (storm %in% c(1, length(names(data)))))
+  }else{
+    stop("Invalid storm input")
+  }
+  
+  # Checking var input
+  stopifnot("invalid var input, must be either 'speed' or 'direction'" = 
+              var == 'speed' | var == 'direction')
+
+}
+
+
+
+
+#' Plotting wind behaviour time series and summary statistics at given point
+#' locations
+#'
+#' @param data time series generated with `temporalBehaviour` with `product=TS`
+#'   input
+#' @param storm list of characters. Storm names. The storm must be available in
+#'   `data` input. It can also be a vector of integer corresponding to the
+#'   indices of storms stored in `data`input.
+#' @param var character. Represent the type of variable to plot. Must be either
+#'   `speed` or `direction`. Default value is set to `speed`.
+#'
+#' @return null
+#' @export
+#'
+#' @examples
+#' \donttest{
+#' #Add example here
+#' }
+plotTemporal <- function(data, storm, var = 'speed'){
+  
+  checkInputsPlotTemporal(data, storm, var)
+  
+  # Filter by storm
+  sub_data <- data[storm]
+  
+  nbOfPositions <- length(sub_data[[1]])
+  
+  # Generate a sequence of colors
+  
+  # TODO pb of dependencies here...
+  palette <- colorRampPalette(colors=c("red", "green","blue"))
+  cols <- palette(nbOfPositions)
+  
+  # Generate a sequence of symbols 
+  symbols <- seq(1, nbOfPositions, 1)
+  
+  # Generate range of wind speed
+  if(var=="speed"){
+    minValue <- c()
+    maxValue <- c()
+    for(location in sub_data[[1]]){
+      minValue <- c(minValue, min(location$speed, na.rm=T))
+      maxValue <- c(maxValue, max(location$speed, na.rm=T))
+    }
+    range <- c(floor(min(minValue))-10, ceiling(max(maxValue) + 10))
+    dy = 5
+  }else{
+    range = c(0, 360)
+    dy = 15
+  }
+  
+  # Plot setting depending on the var input
+  if(var=="speed"){
+    dat <- sub_data[[1]][[1]]$speed
+    ylim <- c(0, 60) # Changer ici avec le range dynamique
+    ylab <- "Maximum Sustained Wind speed (m/s)"
+  }else{
+    dat = sub_data[[1]][[1]]$direction
+    ylab <- "Wind direction (°)"
+    ylim <- c(0, 360)
+  }
+  
+  plot(dat,
+       type = "l",
+       ylim = ylim,
+       xlab = "",
+       ylab = ylab, 
+       axes = FALSE,
+       col = cols[1])
+  
+  graphics::points(dat, col = cols[1], pch = symbols[1])
+  
+  i = 2
+  for(location in sub_data[[1]][c(2:nbOfPositions)]){
+    
+    if(var=="speed"){
+      dat <- location$speed
+    }else{
+      dat = location$direction
+    }
+    
+    graphics::lines(dat, col = cols[i])
+    graphics::points(dat, col = cols[i], pch = symbols[i])
+    i <- i + 1
+  }
+  
+  graphics::legend("topleft",
+         pch = symbols,
+         col = cols,
+         legend = paste(names(sub_data), names(sub_data[[1]])),
+         bty = "n")
+  graphics::axis(1,
```

Comment:
If the length of the storm is too long, the dates are getting messy. It would be good to only display up to a certain amount of dates

Reply:
If there is too much storms, the user can plot TS one by one. I think this new feature in your suggestion would imply too much dev for just a particular case

Reply:
I am not sure you got my point : I am talking about the display of the dates in the x-axis. If one storm lasts e.g. 20 days, and we have information every 3h, then we have 8x20 = 160 dates displayed on the x-axis, which makes the thing quite unreadable.

Reply:
Done in last commit

Reply:
If I do the example plot of the function, I still have ticks and date written every 30mn. DO you have the same behaviour ?

Reply:
I musunderstood the problem here I think, I removes all NA values to shrink the x-axis (which is still a good thing to my mind), but I did not change the setting for the display on the x-axis, I m on it

Reply:
Done,
I modified the code to plot the labels every hours (handling cases of 15min-30min or 1h interpolations)